### DISCUSSION TITLE
Working with Custom dataset and i g

### DISCUSSION THREAD

**Discussion by HackerHakcer** [open-mmlab/OpenLidarPerceptron#1340] [Q&A] [answered]
![screenshot](https://github.com/open-mmlab/OpenPCDet/assets/87998518/bef4fe9e-4bfc-4832-8db3-609215af9fb2)
i generated my own dataset and i customized like required but when i try to generate the infos of the data with this command "python -m pcdet.datasets.custom.custom_dataset create_custom_infos tools/cfgs/dataset_configs/custom_dataset.yaml" 

i got this error  in the screenshot

**Comment by jihanyang** [comment, marked as answer]
Would you like to print the shape of the boxes before the assert error?

**Reply by HackerHakcer** [reply]
thank you @jihanyang the problem indeed was the the shape and it is fixed
My dataset is fine and i generated the data infos successfully 
i have actually another question if it doesn't bother.
I have to train pvrcnn or pvrcnn++ on my custom data but i can't find guided tutorial to follow.
can you please break down the process for me

**Reply by jihanyang** [reply]
You can first select a template from KITTI or waymo (for example, your dataset has front view only or ring view.). Then just replace the DATA_CONFIG to your custom_dataset and follow other settings from PVRCNN++ in kitti or waymo as a baseline.

**Reply by HackerHakcer** [reply]
@jihanyang  your guidance was so helpful thank you
i am now training PVRCNN on my custom dataset and i am wondering if you can give me some heads up about the testing process i want to know the output and the metrics used
